sympy/matrices/matrices.py

```diff
@@ -189,6 +189,9 @@ def __neg__(self):
         return -1*self
 
     def _matrix_pow_by_jordan_blocks(self, num):
+        if (num < 0) == True and self.det() == 0:
```

Comment:
The determinant will be found when the Jordan blocks are computed. Would it be possible to use that instead of another computation here?